External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -305,14 +310,8 @@
 end
 
 % We now subtract the voxel-size adjusted relative coordinate difference from these newly filled voxels
-IM(:, :, :, 1) = IM(:, :, :, 1) + distX(:, :, :)*VoxelSize(1);
-IM(:, :, :, 2) = IM(:, :, :, 2) + distY(:, :, :)*VoxelSize(2);
-IM(:, :, :, 3) = IM(:, :, :, 3) + distZ(:, :, :)*VoxelSize(3);
-
-%% Smoothing of voxels
```

Comment:
Why did you remove this? Any residual NaNs inside the flowfield will corrupt the processing. So we need to fill them.

Reply:
The reason for that was that any NaNs inside the flowfield (e.g. the center of the flowfield) potentially need a different treatment than NaNs at the border of the flowfield. So it was on purpose that the old code "left some NaNs".

Can you confirm what happens with NaNs inside (e.g. the center) of the flowfield in your code?

Reply:
NaNs inside and NaNs outside receive the exact same treatment. We look at nonNaNs, remove unsafe border, extrapolate NaNs linearly with values from the closest safe border.

Previously, with the complicated code, we were scanning each line separately going from image edges from both sides to the border. So that way, we didn't get into the central NaNs. The new solution is much easier and faster and simply looks for the closest border and works with that.

So with a much faster and simpler code, we deal the same way with all NaNs. And the nice thing is that previously, we scanned everything linewise, so the closest border was identified across line only and could lead to weird artifacts. Now, we look for the closest 3D Euclidean distance, so this hsould work much much better.

This was the main reason why I wanted to discard the old line-wise code and replace it.